File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/repository/custom/CustomGenericRequeryRepositoryFactoryBean.java
package com.coupang.springframework.data.requery.repository.custom;

import com.coupang.springframework.data.requery.core.RequeryOperations;
import com.coupang.springframework.data.requery.repository.RequeryRepository;
import com.coupang.springframework.data.requery.repository.support.RequeryRepositoryFactoryBean;
import org.springframework.data.repository.core.support.RepositoryFactorySupport;

import java.io.Serializable;

/**
 * com.coupang.springframework.data.requery.repository.custom.CustomGenericRequeryRepositoryFactoryBean
 *
 * @author debop
 * @since 18. 6. 9
 */
public class CustomGenericRequeryRepositoryFactoryBean<T extends RequeryRepository<Object, Serializable>>
    extends RequeryRepositoryFactoryBean<T, Object, Serializable> {

    public CustomGenericRequeryRepositoryFactoryBean(Class<? extends T> repositoryInterface) {
        super(repositoryInterface);
    }

    @Override
    protected RepositoryFactorySupport createRepositoryFactory(RequeryOperations requeryOperations) {
        return new CustomGenericRequeryRepositoryFactory(requeryOperations);
    }
}

File: spring-data-requery-kotlin/src/test/kotlin/org/springframework/data/requery/kotlin/repository/custom/CustomGenericRequeryRepositoryFactoryBean.kt
package org.springframework.data.requery.repository.custom

import org.springframework.data.requery.kotlin.core.RequeryOperations
import org.springframework.data.requery.repository.RequeryRepository
import org.springframework.data.requery.repository.support.RequeryRepositoryFactory
import org.springframework.data.requery.repository.support.RequeryRepositoryFactoryBean

/**
 * org.springframework.data.requery.repository.custom.CustomGenericRequeryRepositoryFactoryBean
 *
 * @author debop
 */
class CustomGenericRequeryRepositoryFactoryBean<T: RequeryRepository<Any, Any>>(repositoryInterface: Class<out T>)
    : RequeryRepositoryFactoryBean<T, Any, Any>(repositoryInterface) {

    override fun createRepositoryFactory(operations: RequeryOperations): RequeryRepositoryFactory {
        return CustomGenericRequeryRepositoryFactory(operations)
    }
}